testing/fil_token_integration/tests/transfer_tokens.rs

```diff
@@ -0,0 +1,177 @@
+            sequence: unsafe { SEQUENCE },
+            params: if let Some(params) = params { params } else { RawBytes::default() },
+            ..Message::default()
+        };
+        unsafe {
+            SEQUENCE += 1;
+        }
+        self.executor.as_mut().unwrap().execute_message(message, ApplyKind::Explicit, 100).unwrap()
+    }
+
+    fn check_balance(
+        &mut self,
+        operator: Address,
+        token_actor: Address,
+        target: Address,
+    ) -> TokenAmount {
+        let params = RawBytes::serialize(target).unwrap();
+        let ret_val =
+            self.call_method(operator, token_actor, method_hash!("BalanceOf"), Some(params));
+        println!("balance return data {:#?}", &ret_val);
+        ret_val.msg_receipt.return_data.deserialize::<TokenAmount>().unwrap()
+    }
+}
+
+#[test]
+fn transfer_tokens() {
+    let blockstore = MemoryBlockstore::default();
+    let bundle_root = bundle::import_bundle(&blockstore, actors_v10::BUNDLE_CAR).unwrap();
+    let mut tester =
+        Tester::new(NetworkVersion::V15, StateTreeVersion::V4, bundle_root, blockstore.clone())
+            .unwrap();
+
+    let operator: [Account; 1] = tester.create_accounts().unwrap();
+
+    // token actor
+    let token_bin = load_actor_wasm(BASIC_TOKEN_ACTOR_WASM);
+    // transfer actor
+    let transfer_bin = load_actor_wasm(BASIC_TRANSFER_ACTOR_WASM);
+    // account actors
+    let receiver_bin = load_actor_wasm(BASIC_RECEIVER_ACTOR_WASM);
+
+    let token_state = TokenState::new(&blockstore).unwrap();
+    let token_cid = tester.set_state(&token_state).unwrap();
+
+    // transfer actor state
+    let transfer_state = TransferActorState { operator_address: None, token_address: None };
+    let transfer_cid = tester.set_state(&transfer_state).unwrap();
+
+    let token_address = Address::new_id(10000);
+    let transfer_address = Address::new_id(10010);
+    let receiver_address = Address::new_id(10020);
+    tester.set_actor_from_bin(&token_bin, token_cid, token_address, TokenAmount::zero()).unwrap();
+    tester
+        .set_actor_from_bin(&transfer_bin, transfer_cid, transfer_address, TokenAmount::zero())
+        .unwrap();
+    tester
+        .set_actor_from_bin(&receiver_bin, Cid::default(), receiver_address, TokenAmount::zero())
+        .unwrap();
+
+    // Instantiate machine
+    tester.instantiate_machine(DummyExterns).unwrap();
+
+    // construct actors
+    let ret_val =
+        tester.call_method(operator[0].1, token_address, method_hash!("Constructor"), None);
+    println!("token actor constructor return data: {:#?}", &ret_val);
+
+    let ret_val =
+        tester.call_method(operator[0].1, transfer_address, method_hash!("Constructor"), None);
+    println!("transfer actor constructor return data: {:#?}", &ret_val);
+
+    let ret_val =
+        tester.call_method(operator[0].1, receiver_address, method_hash!("Constructor"), None);
+    println!("receiving actor constructor return data: {:#?}", &ret_val);
```

Comment:
There's a lot of boilerplate here that makes it hard to identify the important code. Could you factor some out so that the core test is more to the point?

Reply:
Will follow this up in #103 as it already has a bunch of improvements to the setup boilerplate code